icu4j/main/classes/core/src/com/ibm/icu/impl/breakiter/MlBreakEngine.java

```diff
@@ -0,0 +1,434 @@
+// © 2022 and later: Unicode, Inc. and others.
+// License & terms of use: http://www.unicode.org/copyright.html
+
+package com.ibm.icu.impl.breakiter;
+
+import static com.ibm.icu.impl.CharacterIteration.current32;
+import static com.ibm.icu.impl.CharacterIteration.next32;
+
+import com.ibm.icu.impl.Assert;
+import com.ibm.icu.impl.ICUData;
+import com.ibm.icu.lang.UCharacter;
+import com.ibm.icu.text.UnicodeSet;
+import com.ibm.icu.util.UResourceBundle;
+import com.ibm.icu.util.UResourceBundleIterator;
+
+import java.lang.System;
+import java.text.CharacterIterator;
+import java.util.ArrayList;
+import java.util.HashMap;
+
+public class MlBreakEngine {
+
+    private static final int INVALID = 0x7c; // '|';
+    private static final int MAX_FEATURE = 26;
+    private UnicodeSet fDigitOrOpenPunctuationOrAlphabetSet;
+    private UnicodeSet fClosePunctuationSet;
+    private HashMap<String, Integer> fModel;
+
+    private int fNegativeSum;
+
+    static class Element {
+        private int character;
+        private char[] ublock;
+
+        /**
+         * Default constructor.
+         */
+        public Element() {
+            character = 0;
+            ublock = null;
+        }
+
+        /**
+         * Set the character and its unicode block.
+         *
+         * @param ch  A unicode character.
+         * @param str The unicode block of the character.
+         */
+        public void setCharAndUblock(int ch, String str) {
+            Assert.assrt(str.length() <= 3);
+            this.character = ch;
+            ublock = str.toCharArray();
```

Comment:
In Java, this does not really help: The char array is a new heap-allocated Object just like the incoming String.

I suggest that you change ublock to just store a String.
It's also ok if you don't want to change this now.

----
It *might* reduce object creations if we worked with a mutable StringBuilder, or something similar, but let's not try that in this PR.

Reply:
Done